File: cmd/scaffold.go
package cmd

import (
	"fmt"
	"os"

	"github.com/spf13/cobra"
)

func createDefaultScript(name string) error {
	file, err := os.Create(name)
	if err != nil {
		return fmt.Errorf("Failed to create init/%s/init.sh: %v", name, err)
	}

	if err := WriteToFile(file, []string{
		"#!/bin/bash",
		"",
	}); err != nil {
		return fmt.Errorf("Failed to initialize %s: %v", name, err)
	}

	return nil
}

func scaffold(name string) error {
	if err := os.Mkdir(fmt.Sprintf("init/%s", name), 0777); err != nil {
		return fmt.Errorf("Failed to create init/%s: %v", name, err)
	}

	if err := createDefaultScript(fmt.Sprintf("init/%s/init.sh", name)); err != nil {
		return err
	}
	if err := createDefaultScript(fmt.Sprintf("init/%s/install.sh", name)); err != nil {
		return err
	}

	return nil
}

var scaffoldCmd = &cobra.Command{
	Use:   "scaffold",
	Short: "Init script scaffolder",
	Long:  `Init script scaffolder`,
	Args:  cobra.ExactArgs(1),
	RunE: func(cmd *cobra.Command, args []string) error {
		err := scaffold(args[0])
		if err != nil {
			return fmt.Errorf("Failed to scaffold init files: %v", err)
		}
		return nil
	},
}

func init() {
	rootCmd.AddCommand(scaffoldCmd)
}

File: linux/src/cmds/scaffold.rs
use std::{
    fs::{create_dir, File},
    io::Write,
    path::Path,
};

use crate::error::{Error, Result};

fn create_default_script(name: &str) -> Result<()> {
    let file_path = Path::new(name);
    let mut file = File::create(file_path).map_err(Error::IOError)?;

    file.write_all(b"#!/bin/bash\n\n").map_err(Error::IOError)?;

    Ok(())
}

pub fn run(name: &str) -> Result<()> {
    let dir_path = format!("init/{}", name);
    create_dir(&dir_path).map_err(Error::IOError)?;

    create_default_script(&format!("{}/init.sh", dir_path))?;
    create_default_script(&format!("{}/install.sh", dir_path))?;

    Ok(())
}
